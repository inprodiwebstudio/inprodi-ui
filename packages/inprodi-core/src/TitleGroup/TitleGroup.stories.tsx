import React                             from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { TitleGroup, TitleGroupProps }   from "./TitleGroup";

export default {
	title     : "Core/Typography/TitleGroup",
	component : TitleGroup,
	argTypes  : {
		size        : { control : "text" },
		textColor   : { control : "color" },
		titleColore : { control : "color" },
	},
} as ComponentMeta<typeof TitleGroup>;

const Template: ComponentStory<typeof TitleGroup> = (args : TitleGroupProps) => {

	return(
		<TitleGroup {...args} />
	);
};

export const Basic = Template.bind({});
Basic.args = {
	title : "This is a title",
	text  : "This is a subtext",
	size  : "lg",
};
