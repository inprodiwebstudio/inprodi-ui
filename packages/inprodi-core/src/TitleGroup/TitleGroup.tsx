import {
	Text,
	Title,
	Stack,
	TextProps,
	StackProps,
	MantineNumberSize,
	DefaultMantineColor,
} from "@mantine/core";
import { TitleProps } from "../Title/Title";

export interface TitleGroupProps extends StackProps {
    text : string;
    title : string;
    textProps ?: TextProps;
    titleProps ?: TitleProps;
    size ?: MantineNumberSize;
    textColor ?: DefaultMantineColor;
    titleColor ?: DefaultMantineColor;
}

export const TitleGroup = ({
	text,
	title,
	textColor,
	titleColor,
	titleProps,
	textProps,
	size = "md",
	...rest
} : TitleGroupProps) => {
	return (
		<Stack data-testid="stack-group-stack" spacing={0} {...rest}>
			<Title data-testid="title-group-stack" size={size} color={titleColor} {...titleProps}>{title}</Title>
			<Text data-testid="text-group-stack" size={size} color={textColor} {...textProps}>{text}</Text>
		</Stack>
	);
};

export default TitleGroup;
