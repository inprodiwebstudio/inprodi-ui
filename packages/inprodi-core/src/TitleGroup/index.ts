export { TitleGroup } from "./TitleGroup";
export type { TitleGroupProps } from "./TitleGroup";
