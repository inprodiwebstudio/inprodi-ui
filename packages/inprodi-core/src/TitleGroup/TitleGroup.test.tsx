import React          from "react";
import { render }     from "@testing-library/react";
import { TitleGroup } from "./TitleGroup";

describe("TitleGroup", () => {
	test("snapshot", () => {
		const { container } = render(
			<TitleGroup
				title="Title"
				text="text"
			/>
		);
		expect(container).toMatchSnapshot();
	});
	test("should has all the correct classNames", () => {
		const stackClassName = "stack-className";
		const titleClassName = "title-className";
		const textClassName = "text-className";
		const { getByTestId } = render(
			<TitleGroup
				spacing={15}
				className={stackClassName}
				title="Title"
				text="text"
				textProps={{
					className : textClassName,
				}}
				titleProps={{
					className : titleClassName,
				}}
			/>
		);
		expect(getByTestId("stack-group-stack").className.includes(stackClassName)).toBeTruthy();
		expect(getByTestId("title-group-stack").className.includes(titleClassName)).toBeTruthy();
		expect(getByTestId("text-group-stack").className.includes(textClassName)).toBeTruthy();
	});
});
