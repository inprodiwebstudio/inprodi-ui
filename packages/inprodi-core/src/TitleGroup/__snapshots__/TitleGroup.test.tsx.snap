// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleGroup snapshot 1`] = `
<div>
  <div
    class="mantine-Stack-root mantine-1fod3h5"
    data-testid="stack-group-stack"
  >
    <h1
      class="mantine-Text-root mantine-Title-root mantine-73tske"
      data-testid="title-group-stack"
    >
      Title
    </h1>
    <div
      class="mantine-Text-root mantine-qls5vf"
      data-testid="text-group-stack"
    >
      text
    </div>
  </div>
</div>
`;
